src/components/MiniWidgetContainer.vue

```diff
@@ -115,4 +117,13 @@ const refreshWidgetsHashs = (): void => {
 const showWidgetTrashArea = ref(false)
 
 const trashList = ref<MiniWidget[]>([])
+
+const handleDeleteWidget = (event: DraggableEvent): void => {
+  const widgetData = container.value.widgets.find((w) => w.hash === event.item.dataset.widgetHash)
+  if (widgetData) {
+    // Remove miniWidget variableName from Logged variables list
+    CurrentlyLoggedVariables.removeVariable(widgetData.options.displayName)
+  }
+  trashList.value = []
+}
 </script>
```

Comment:
Same here as the last commit. This is specific to the VGI widget, but we are coupling it with the MiniWidget instantiation system.

Reply:
For the time, so we do not hurt the roadmap, we decide to let this coupling how it is and open an issue to track a future improvement on it.